test/e2e/framework/storageclass_util.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 Oracle and/or its affiliates. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CheckStorageClass verifies the storage class exists, if not creates a storage class
+func (j *PVCTestJig) CheckStorageClass(name string) bool {
+	list, err := j.KubeClient.StorageV1beta1().StorageClasses().List(metav1.ListOptions{})
+	if err != nil {
+		Failf("Error listing storage classes: %v", err)
+	}
+
+	for _, sc := range list.Items {
+		if sc.Name == name {
+			Logf("Storage class %q found", sc.Name)
+			return true
+		}
+	}
+
+	return false
+}
+
+// CheckSCorCreate checks if a storage class exists, if not creates one
+func (j *PVCTestJig) CheckSCorCreate(name string, provisionerType string, param map[string]string) {
```

Comment:
Rather than doing a check then create I'd suggest doing a Create, checking the returned error to see if its `apierrors.IsAlreadyExists(err)`, and then Updating if it is. That way we always ensure that the StorageClass looks like we expect (at least at the point of its creation/updating).

(`import apierrors "k8s.io/apimachinery/pkg/api/errors"`).